src/nvim/normal.c

```diff
@@ -1711,11 +1711,11 @@ void do_pending_operator(cmdarg_T *cap, int old_col, bool gui_yank)
        */
       if (oap->motion_force == NUL || oap->motion_type == MLINE)
         oap->inclusive = true;
-      if (VIsual_mode == 'V')
+      if (VIsual_mode == 'V') {
         oap->motion_type = MLINE;
-      else {
+      } else if (VIsual_mode == 'v') {
         oap->motion_type = MCHAR;
-        if (VIsual_mode != Ctrl_V && *ml_get_pos(&(oap->end)) == NUL
+        if (*ml_get_pos(&(oap->end)) == NUL
```

Comment:
Before these changes, `oap->motion_type = MCHAR` was always set. Why is it safe to not do that now?